server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/CellIdSource.java

```diff
@@ -81,20 +87,32 @@ public SortedBinaryDocValues bytesValues(LeafReaderContext ctx) {
         private MultiGeoPointValues geoValues;
         private int precision;
         private GeoPointLongEncoder encoder;
+        private org.elasticsearch.common.geo.GeoPoint topLeft;
+        private org.elasticsearch.common.geo.GeoPoint bottomRight;
 
-        protected CellValues(MultiGeoPointValues geoValues, int precision, GeoPointLongEncoder encoder) {
+        protected CellValues(MultiGeoPointValues geoValues, int precision, org.elasticsearch.common.geo.GeoPoint topLeft,
+                             org.elasticsearch.common.geo.GeoPoint bottomRight, GeoPointLongEncoder encoder) {
             this.geoValues = geoValues;
             this.precision = precision;
             this.encoder = encoder;
+            this.topLeft = topLeft;
+            this.bottomRight = bottomRight;
+            assert Double.isNaN(topLeft.lon()) ==
+                ((Double.isNaN(topLeft.lat()) ^ Double.isNaN(bottomRight.lon()) ^ Double.isNaN(bottomRight.lat())));
         }
 
         @Override
         public boolean advanceExact(int docId) throws IOException {
             if (geoValues.advanceExact(docId)) {
                 resize(geoValues.docValueCount());
+                int j = 0;
                 for (int i = 0; i < docValueCount(); ++i) {
                     org.elasticsearch.common.geo.GeoPoint target = geoValues.nextValue();
-                    values[i] = encoder.encode(target.getLon(), target.getLat(), precision);
+                    if (Double.isNaN(topLeft.lon()) ||
+                            (target.lon() < bottomRight.lon() && target.lon() > topLeft.lon()
+                            && target.lat() < bottomRight.lat() && target.lat() > topLeft.lat())) {
+                        values[j++] = encoder.encode(target.getLon(), target.getLat(), precision);
```

Comment:
hey @iverase. I thought it a good idea to introduce this `bounds` idea directly to geotile_grid in master first.
The check has some work to do with regards to normalization and dateline handling I believe. I was wondering if there are any geo-utils already around for checking if a point is in a bounding box that I should be re-using here.

Reply:
Nothing that is worthy.

Reply:
OK. I wrote something and added it to GeoUtils